### PR TITLE
First round of change proposals on multiple host support

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

File: src/Npgsql/NpgsqlConnector.cs [hunk 1]
```diff
@@ -564,13 +566,9 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
             TypeMapper.Bind(DatabaseInfo);
         }
 
-        internal async ValueTask UpdateServerType(NpgsqlTimeout timeout, bool async,
-            CancellationToken cancellationToken = default)
+        internal async ValueTask<ClusterState> UpdateServerType(
+            NpgsqlTimeout timeout, bool async, CancellationToken cancellationToken = default)
         {
-            // Extended query protocol not supported in a replication connection
```
Comment (vonzshik) [inline src/Npgsql/NpgsqlConnector.cs:570]: That's definitely not something we should remove, as otherwise it fails for the replication connection.

Comment (roji) [reply src/Npgsql/NpgsqlConnector.cs:570]: I've added validation to disallow multiple hosts for replication, just like for multiplexing - at least for now. I'm not sure of the possible consequences. IIRC we don't even support pooling with replication connections...

Comment (roji) [reply src/Npgsql/NpgsqlConnector.cs:570]: Oh, the tests have helped me understand :) Brought back the check.

**Review by vonzshik — APPROVED** [review]
This actually looks pretty cool!

**Pull request merged**